mlir/include/mlir/Dialect/LLVMIR/GENXOps.td

```diff
@@ -338,7 +338,64 @@ def GENX_AtomicRMWOp : GENX_Op<"atomic.rmw", [
 //===----------------------------------------------------------------------===//
 // Joint Matrix operations
 //===----------------------------------------------------------------------===//
+
 def GENX_MatrixElemType : AnyTypeOf<[AnyI8, AnyI16, AnyI32, F32, F16, BF16]>;
+
+def GENX_MatrixDPASOp : GENX_Op<"matrix.dpas">,
+  Results<(outs FixedVectorOf<[GENX_MatrixElemType]>:$d)>,
+  Arguments<(ins 
+    FixedVectorOf<[GENX_MatrixElemType]>:$c,
+    FixedVectorOf<[GENX_MatrixElemType]>:$a,
+    FixedVectorOf<[GENX_MatrixElemType]>:$b,
+    I32Attr:$pa,
+    I32Attr:$pb,
+    I32Attr:$sd,
+    I32Attr:$rc,
+    I1Attr:$isDpasw
+  )> {
+
+  let summary = "GEN matrix multiply-add";
+
+  string baseDescription = [{
+    The 'genx.matrix.dpas' operation is a matrix multiply-add operation as follows:
+
+       D = C + A x B
+
+       where
+          D : MxN
+          C : MxN
+          A : MxK
+          B : KxN
+
+          M : repeat count ($rc)
+          N : fixed execution size, either 8 or 16
```

Comment:
This has to be 16 for the PVC architecture and has to be 8 for previous GPU generations. So the question is  whether we want to support just PVC or not. My opinion is we make this mandatory to be 16. Propose to change this text to:
```
  N : fixed execution size, must be 16 for PVC